lib/spack/spack/build_environment.py

```diff
@@ -222,10 +225,47 @@ def set_build_environment_variables(pkg, env, dirty):
     build_link_deps = build_deps | link_deps
     rpath_deps      = get_rpath_deps(pkg)
 
+    link_dirs = []
+    include_dirs = []
+    rpath_dirs = []
+
+    # Set up link, include, RPATH directories that are passed to the
+    # compiler wrapper
+    for dep in link_deps:
+        if is_system_path(dep.prefix):
+            continue
+        # TODO: packages with alternative implementations of .libs which
+        # are external may place libraries in nonstandard directories, so
+        # there should be a check for that
+        query = pkg.spec[dep.name]
+        try:
+            dep_link_dirs = list(query.libs.directories)
+            link_dirs.extend(dep_link_dirs)
+            if dep in rpath_deps:
+                rpath_dirs.extend(dep_link_dirs)
+        except spack.spec.NoLibrariesError:
+            tty.debug("No libraries found for {0}".format(dep.name))
+
+        try:
+            include_dirs.extend(query.headers.directories)
+        except spack.spec.NoHeadersError:
+            tty.debug("No headers found for {0}".format(dep.name))
+        if os.path.isdir(pkg.prefix.include):
```

Comment:
Since this is the include dir for the dependencies, shouldn't this be `query.prefix.include` ?

Making this change lets gdbm install, without the change, it still fails to include the readline headers